src/main/java/se/sundsvall/casedata/service/scheduler/emailreader/EmailReaderWorker.java

```diff
@@ -47,34 +51,44 @@ public EmailReaderWorker(final MessageRepository repository, final ErrandReposit
 		this.notificationService = notificationService;
 	}
 
-	void getAndProcessEmails() {
+	List<Email> getEmails() {
 
 		try {
-			emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace())
-				.forEach(this::saveAndRemoteDelete);
+			return emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace());
+
 		} catch (final Exception e) {
 			LOG.error("Error when fetching emails from EmailReader", e);
 		}
+		return emptyList();
 	}
 
-	@Transactional
-	public void saveAndRemoteDelete(final Email email) {
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public boolean save(final Email email) {
 		try {
 			final var errandNumber = parseSubject(email.getSubject());
 
 			errandRepository.findByErrandNumber(errandNumber)
 				.filter(errand -> !messageRepository.existsById(email.getId()))
 				.ifPresent(errand -> {
+					Hibernate.initialize(errand.getStakeholders());
 					messageRepository.save(emailReaderMapper.toMessage(email, errand.getMunicipalityId(), errand.getNamespace()).withErrandNumber(errandNumber));
 					notificationService.createNotification(errand.getMunicipalityId(), errand.getNamespace(), toNotification(errand, NOTIFICATION_TYPE, NOTIFICATION_DESCRIPTION));
 					attachmentRepository.saveAll(emailReaderMapper.toAttachments(email, errand.getMunicipalityId(), errand.getNamespace()).stream()
 						.map(attachment -> attachment.withErrandNumber(errandNumber).withMunicipalityId(emailReaderProperties.municipalityId()))
```

Comment:
`.withMunicipalityId` is redundant here, right?

Reply:
yup